internal/config/config.go

```diff
@@ -170,44 +170,68 @@ func For(resources ClusterResources, validate Validate) (*Config, error) {
 		return nil, err
 	}
 
-	cfg := &Config{
-		Pools:       map[string]*Pool{},
-		BFDProfiles: map[string]*BFDProfile{},
+	cfg := &Config{}
+
+	cfg.BFDProfiles, err = forBFDProfiles(resources)
+	if err != nil {
+		return nil, err
+	}
+
+	cfg.Peers, err = forPeers(resources, cfg.BFDProfiles)
+	if err != nil {
+		return nil, err
+	}
+
+	cfg.Pools, err = forPools(resources)
+	if err != nil {
+		return nil, err
 	}
 
+	return cfg, nil
+}
+
+func forBFDProfiles(resources ClusterResources) (map[string]*BFDProfile, error) {
```

Comment:
I'd move this (and the others) as `bfdProfilesFor`, sorry I was not clear with my previous comment

Reply:
np, fixed.